lib/gems/pending/util/mount/miq_generic_mount_session.rb

```diff
@@ -464,6 +467,10 @@ def self.raw_disconnect(mnt_point)
     end
   end
 
+  def copy_dump_to_store
```

Comment:
Would it not be better to override `#add`/`#remove`/`#download` rather than creating a specific method for this which none of the other sub-classes will use?

Reply:
Well, I'm not against overriding if it makes sense, but as far as none of the other sub-classes implementing this method don't I have to do the same thing for OpenStack Swift? And any other object-store implementation?

Reply:
Well, yeah, that's kind of what I'm saying. I feels like the API for an instance of `MiqGenericMountSession` is supposed to be `#add`/`#download`/ etc. so if we don't have an alternate implementation for add for object stores, it feels like this fits the bill.

The only reason I though about this was that `copy_dump_to_store` isn't really a good name, right? Because you could be copying any file to the store, not just a dump. And that got me thinking that this is really just the implementation of `#add` for things which don't actually "mount".

Does that make sense?

Reply:
@carbonin note that I've reworked this area somewhat to handle your request and some of the errors encountered.  I still have to push out the Core Repo changes although those might get modified based on UI changes as well.

Reply:
I think this can be removed now, right?